client/object_put.go

```diff
@@ -275,12 +273,12 @@ func (x *objectWriter) InitDataStream(header object.Object) (io.Writer, error) {
 		}, nil
 	}
 
-	res, err := stream.Close()
+	_, err = stream.Close()
 	if err != nil {
 		return nil, err
 	}
 
-	return nil, apistatus.ErrFromStatus(res)
+	return nil, nil
```

Comment:
`stream.Close()` returns [ResObjectPut](https://github.com/nspcc-dev/neofs-sdk-go/blob/7002b3b0df27a4f1daa9e8ed7cb8edfa69c2a355/client/object_put.go#L38) which doesn't have error. It always will be `nil`

Reply:
i think that's not a rule but a _too-generic_ code instead. i can imagine it will have its own statuses

Reply:
But this means that `WriteHeader` has failed and technically `err` can't be nil. We can't return `nil` error in this case from `InitDataStream` as well, it'll be interpreted as success and caller will try using `io.Writer` just to fail (with a panic!) immediately.

Reply:
If it was failed, the error will be returned few lines upper - I mean line 278:
```go
	if err != nil {
		return nil, err
	}
```

Reply:
I mean this path is an error handler already, `WriteHeader()` has to return `false` (fail!) and we're just trying to retrieve this error via `Close()` and return it to the caller. But there is an error. If `Close()` doesn't return an error after unsuccessful `WriteHeader()`, it's a bug in SDK. `return nil, nil` is not an option in this case, better panic right here than in the client code. `return nil, stream.err` is also questionable. `return nil, errors.New("i have no idea how this has happened")` can be more appropriate.

Reply:
Ok, I see. Replaced with `errors.New("unexpected error")`